sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-queue-retry.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,230 @@
+    });
+
+    it("Should queue an assessment for cancellation with 'cancelation queued' status when it has this status for more than the time configured for the queue.", async () => {
+      // Arrange
+      const application = await saveFakeApplication(db.dataSource);
+      const currentAssessment = application.currentAssessment;
+      currentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.CancellationQueued;
+      // Gets one hour more than the needed time for the assessment retry.
+      currentAssessment.updatedAt =
+        await getAmountOfHoursAssessmentRetryWithHourOffset(1);
+      await db.studentAssessment.save(currentAssessment);
+      // Retry job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await retryProcessor.enqueueAssessmentRetryOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+      // Assert item was added to the queue.
+      const queueData = {
+        assessmentId: updatedApplication.currentAssessment.id,
+      } as CancelAssessmentQueueInDTO;
+      expect(cancelApplicationAssessmentQueueMock.add).toBeCalledWith(
+        queueData,
+      );
+    });
+
+    it("Should not queue an assessment with 'queued' status when it has this status for less than the time configured for the queue.", async () => {
+      // Arrange
+      const application = await saveFakeApplication(db.dataSource);
+      const currentAssessment = application.currentAssessment;
+      currentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.Queued;
+
+      // Gets one hour less than the needed time for the assessment retry.
+      currentAssessment.updatedAt =
+        await getAmountOfHoursAssessmentRetryWithHourOffset(-1);
+      await db.studentAssessment.save(currentAssessment);
+
+      // Retry job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await retryProcessor.enqueueAssessmentRetryOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+      // Assert item was added to the queue.
+      const queueData = {
+        assessmentId: updatedApplication.currentAssessment.id,
+      } as StartAssessmentQueueInDTO;
+      expect(startApplicationAssessmentQueueMock.add).not.toBeCalledWith(
+        queueData,
+      );
+    });
+
+    it("Should not queue an assessment cancellation with 'cancellation queued' status when it has this status for less than the time configured for the queue.", async () => {
+      // Arrange
+      const application = await saveFakeApplication(db.dataSource);
+      const currentAssessment = application.currentAssessment;
+      currentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.Queued;
```

Comment:
shouldn't this be `CancellationQueued` ?

Reply:
Good catch. I changed that.